### PR TITLE
feat: support bitwise functions

### LINKED ISSUE #225: Bitwise function: `&, |, ^, ~`
**Is your feature request related to a problem? Please describe.**

Support bitwise operators: `&, |, ^, ~`

### `expr1 & expr2`
 - Returns the result of bitwise AND of expr1 and expr2.

```SQL
-- Examples:

> SELECT 3 & 5;
1
```

### `expr1 ^ expr2` 
- Returns the result of bitwise exclusive OR of expr1 and expr2.

```SQL


Examples:

> SELECT 3 ^ 5;
 6
```

### `expr1 | expr2` 
- Returns the result of bitwise OR of expr1 and expr2.
```SQL
Examples:

> SELECT 3 | 5;
 7
```

### `~ expr `
- Returns the result of bitwise NOT of expr.
```
Examples:

> SELECT ~ 0;
 -1
```
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by aceforeverd**
* **Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

close #225 .

1. support bitwise operators:
  - `&`
  - `|`
  - `^`
  - `~`
2. fix some clang-tidy lint report in codegen
  - avoid use `_` in google test name
  - use static_cast/dynamic_cast/reinterpret_cast instead of C-style convert
  - do not use `using` in header file
3. type infer
  - both `lhs` and `rhs` should be `int16`, `int32`, `int64`
  - nullable

TODO
- [x] update example test

### CODE REVIEW

File: hybridse/src/codegen/arithmetic_expr_ir_builder.cc [hunk 1]
```diff
@@ -442,6 +476,57 @@ Status ArithmeticIRBuilder::BuildModExpr(const NativeValue& left,
...
+            return BuildAnd(block, lhs, rhs, output, status);
+        },
+        output));
+    return Status::OK();
+}
+
+Status ArithmeticIRBuilder::BuildBitwiseOrExpr(const NativeValue& left, const NativeValue& right, NativeValue* output) {
+    CHECK_STATUS(TypeIRBuilder::BinaryOpTypeInfer(
+        node::ExprNode::BitwiseLogicalTypeAccept, left.GetType(), right.GetType()));
+    CHECK_STATUS(NullIRBuilder::SafeNullBinaryExpr(
```
Comment (aceforeverd) [inline hybridse/src/codegen/arithmetic_expr_ir_builder.cc:503]: does safe null builder needed ?

File: cases/function/expression/test_arithmetic.yaml [hunk 2]
```diff
@@ -593,3 +593,107 @@ cases:
            NULL, NULL, NULL, NULL, NULL,
            NULL, NULL, NULL, NULL, NULL, NULL, NULL,
            NULL, NULL, NULL, NULL, NULL]
+  - id: bitwise_operators
+    desc: bitwise and/or/xor
+    inputs:
```
Comment (aceforeverd) [inline cases/function/expression/test_arithmetic.yaml:599]: mode: ['hybridse-only']

File: cases/function/expression/test_arithmetic.yaml [hunk 3]
```diff
@@ -593,3 +593,107 @@ cases:
...
+      - columns: ["c1 int16","c2 int32","c3 bigint", "c4 int16", "c6 timestamp"]
+        indexs: ["index1:c3:c6"]
+        rows:
+          - [3, 6, 12, NULL, 1590738989000]
+    sql: |
+      select {0}.c1 & {0}.c4 as r1, {0}.c2 | {0}.c4 as r2, {0}.c3 ^ {0}.c4 as r3, ~ {0}.c4 as r4 from {0}
+    expect:
+      columns:
+        - r1 int16
```
Comment (aceforeverd) [inline cases/function/expression/test_arithmetic.yaml]: - [ ]

**Review by jingchen2222 — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Review by imotai — APPROVED** [review]

**Pull request merged**